src/pem/_core.py

```diff
@@ -40,6 +40,16 @@ def as_bytes(self):
         """
         return self._pem_bytes
 
+    def __eq__(self, other):
+        return (type(self) == type(other) and
```

Comment:
I pushed a change that returns `NotImplemented`; only for things that aren't subclasses of `_Base`, though. I think it's obvious that `Certificate(...).__eq__(RSAPrivateKey(...))` should be `False` not `NotImplemented`? But let me know if you disagree.